elastic/apm-server
# Update to opentelemetry-collector v0.34.0

## Motivation/summary

Update to opentelemetry-collector v0.34.0.

The Jaeger translator and receiver have moved to opentelemetry-collector-contrib, so our Jaeger code has changed accordingly.

There are no longer distinct Int and Double metric types, but there are now Int and Double numeric data points. Changed our OTLP metrics handling accordingly. This fixes https://github.com/elastic/apm-server/issues/6008.

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

- Instrument an application with https://github.com/open-telemetry/opentelemetry-java-instrumentation, point it at apm-server
- Check that JVM metrics show up in the APM UI

## Related issues

Fixes https://github.com/elastic/apm-server/issues/6008
